test/src/androidTest/kotlin/io/realm/NullabilityTests.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.runtimeapi.RealmModule
+import test.Nullability
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+class NullabilityTests {
+
+    @RealmModule(Nullability::class)
+    class MySchema
+
+    lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        val configuration = RealmConfiguration.Builder(schema = MySchema()).build()
+        realm = Realm.open(configuration)
+        // FIXME Cleaning up realm to overcome lack of support for deleting actual files
+        //  https://github.com/realm/realm-kotlin/issues/95
+        realm.beginTransaction()
+        realm.objects(Nullability::class).delete()
+        realm.commitTransaction()
+        assertEquals(0, realm.objects(Nullability::class).size, "Realm is not empty")
+    }
+
+    @Test
+    fun nullability() {
+        realm.beginTransaction()
+        val nullability = realm.create(Nullability::class)
+        assertNull(nullability.stringNullable)
```

Comment:
Maybe we should disable this assertion until we implement default values? because it's weird to assert for nullability on `var stringNullable: String? = ""` maybe enable this test if the declared property was `var stringNullable: String? = null`